src/algorithm/butterflies.rs

```diff
@@ -430,78 +422,77 @@ impl<T> IsInverse for Butterfly6<T> {
-        let twiddle3: Complex<T> = twiddles::single_twiddle(3, 7, inverse) * sixth;
-
-        //our primitive root will be 3, and our inverse will be 5. the powers of 5 mod 7 are 1,5,4,6,2,3, so we hardcode to use the twiddles in that order
-        let mut fft_data = [twiddle1, twiddle2.conj(), twiddle3.conj(), twiddle1.conj(), twiddle2, twiddle3];
-
-        let butterfly = Butterfly6::new(inverse);
-        unsafe { butterfly.process_inplace(&mut fft_data) };
-
+        let twiddle1: Complex<T> = twiddles::single_twiddle(1, 7, inverse);
+        let twiddle2: Complex<T> = twiddles::single_twiddle(2, 7, inverse);
+        let twiddle3: Complex<T> = twiddles::single_twiddle(3, 7, inverse);
         Butterfly7 { 
-            inner_fft: butterfly,
-            inner_fft_multiply: fft_data,
+            twiddle1, 
+            twiddle2,
+            twiddle3,
+            inverse,
         }
     }
 }
 impl<T: FFTnum> FFTButterfly<T> for Butterfly7<T> {
-    #[inline(always)]
-    unsafe fn process_inplace(&self, buffer: &mut [Complex<T>]) {
-        //we're going to reorder the buffer directly into our scratch vec
-        //our primitive root is 3. use 3^n mod 7 to determine which index to copy from
-        let mut scratch = [
-            *buffer.get_unchecked(3),
-            *buffer.get_unchecked(2),
-            *buffer.get_unchecked(6),
-            *buffer.get_unchecked(4),
-            *buffer.get_unchecked(5),
-            *buffer.get_unchecked(1),
-            ];
-
-        //perform the first inner FFT
-        self.inner_fft.process_inplace(&mut scratch);
-
-        //multiply the fft result with our precomputed data
-        for i in 0..6 {
-            scratch[i] = scratch[i] * self.inner_fft_multiply[i];
-        }
-
-        //perform the second inner FFT
-        let inverse6 = Butterfly6::inverse_of(&self.inner_fft);
-        inverse6.process_inplace(&mut scratch);
-
-        //the first element of the output is the sum of the rest
-        let first_input = *buffer.get_unchecked(0);
-        let mut sum = first_input;
-        for i in 1..7 {
-            sum = sum + *buffer.get_unchecked_mut(i);
-        }
-        *buffer.get_unchecked_mut(0) = sum;
-
-        //use the inverse root ordering to copy data back out
-        *buffer.get_unchecked_mut(5) = scratch[0] + first_input;
-        *buffer.get_unchecked_mut(4) = scratch[1] + first_input;
-        *buffer.get_unchecked_mut(6) = scratch[2] + first_input;
-        *buffer.get_unchecked_mut(2) = scratch[3] + first_input;
-        *buffer.get_unchecked_mut(3) = scratch[4] + first_input;
-        *buffer.get_unchecked_mut(1) = scratch[5] + first_input;
-    }
     #[inline(always)]
     unsafe fn process_multi_inplace(&self, buffer: &mut [Complex<T>]) {
         for chunk in buffer.chunks_mut(self.len()) {
             self.process_inplace(chunk);
         }
     }
+    #[inline(always)]
+    unsafe fn process_inplace(&self, buffer: &mut [Complex<T>]) {
```

Comment:
Could you leave come comments about how this algorithm was derived? I think I know what's happening here -- it looks familiar to me because I recently implemented the same thing in the SIMD branch -- but someone else reading it may not understand.

One of my goals for this project is for this to be something people can refer to when implementing FFTs themselves. I don't think each and every line needs to be commented, but at least linking to a wikipedi page, or telling readers what  algebraic steps will lead them to this set of operations, will go a long way.